### PR TITLE
Add profile to list in upsertProfile

### PR DESCRIPTION
**Pull request by harsh-joshi99**
This PR 

- Adds list Id mapping in upsertProfile action
- Adds profile to list if list Id is provided
- Updates unit tests

JIRA -> [STRATCONN-3358](https://segment.atlassian.net/browse/STRATCONN-3358)

### Testing in stage (running locally via stage)
Screenshot 2023-11-03 at 10 42 58 AM
Screenshot 2023-11-03 at 10 47 25 AM
Screenshot 2023-11-03 at 10 47 57 AM

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

[STRATCONN-3358]: https://segment.atlassian.net/browse/STRATCONN-3358?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/destination-actions/src/destinations/klaviyo/functions.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    ]
+  }
+
+  try {
+    await request(`${API_URL}/lists/${listId}/relationships/profiles/`, {
+      method: 'POST',
+      json: listData
+    })
+  } catch (error) {
+    throw new APIError('An error occurred while processing the request', 400)
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/klaviyo/functions.ts:52]: Is there any reason this error needs to be handled? Wouldn't the default HTTPError be enough here?

**Review by varadarajan-tw — APPROVED** [review]
LGTM! Left a couple of comments but they aren't a blocker. Lets not merge the PR till next Tuesday.

**Comment by joe-ayoub-segment** [comment]
note: Plan to deploy this on Tuesday 14th - will also do a push.

**Comment by joe-ayoub-segment** [comment]
Hi @harsh-joshi99 this PR has been deployed.

**Pull request merged**